site-src/v1alpha2/guides/unmanaged-gateway-development.md

```diff
@@ -0,0 +1,107 @@
+going to inevitably be somewhat esoteric and unique to each individual
+implementation.
+
+In spite of the open-ended nature of unmanaged gateway implementations this
+guide intends to provide some high level guidance for developers on how the
+logic for unmanaged `Gateway` implementations might look in hopes of providing
+a basic level of guidance and in hopes of somewhat reducing the disparity
+between the different implementations out there. Each section below is
+effectively an idea or example for your considerations, but none of this should
+be considered any kind of strict requirement.
+
+## Configuration Idea: Flag Unmanaged With Annotations
+
+Since the `Gateway` spec itself does not include any concept of managed or
+unmanaged gateway, a reasonable approach to indicate on new `Gateway` objects
+that they are unmanaged, and otherwise reference an existing deployment may
+be to require an annotation:
+
+```yaml
+apiVersion: gateway.networking.k8s.io/v1alpha2
+kind: Gateway
+metadata:
+  name: my-gateway
+  annotations:
+    acme.io/provisioning-mode: "unmanaged"
+spec:
+  listeners:
+  - name: http
+    protocol: HTTP
+    port: 80
+```
+
+For implementations which support both managed and unmanaged, this makes it
+very straightforward to branch reconcilation logic based on the mode:
+
+```golang
+mode := "acme.io/provisioning-mode"
+if mode, ok := gateway.Annotations[mode]; ok && mode == "unmanaged" {
+    handleUnmanaged(gateway)
+} else {
+    handleManaged(gateway)
+}
+```
+
+## Controller Idea: Gateway Service Reference
+
+When implementing unmanaged style `Gateway` logic, it may be helpful to start
+by defining a concept of what the `Gateway` really is, or what it represents
+within the context of the existing server it represents.
+
+You may for the purposes of this guide consider a `Gateway` resource to be
+similar to a Kubernetes [Service][svc] resource in that there are `Ports`,
+`Protocols` and `IPs` which are analogs to the `Listeners` and `Addresses`
+within `Gateway`. Conceptually you might consider a `Gateway` as similar to a
+`Service` that provides rules about which controllers should act on it (via
+`GatewayClass`) as well as rules about how other APIs such as `HTTPRoute` may
+attach to it.
+
+While thinking of a `Gateway` as effectively being a `Service` may not be right
+for every implementation, if your implementation does in fact include an
+existing `Service` resource which is distinct and well defined and it might be
+feasible to consider it the "source of truth" for the `Listeners` and
+`Addresses` that would otherwise be represented by the `Gateway`, then perhaps
+consider coupling the two together:
+
+- require a reference to a `Service` to be present for unmanaged `Gateway`
+  functionality to be available, perhaps via annotations or via controller
+  manager command line flags and environment variables.
+- add reconcilation logic which can convert a `Service` spec into `Listeners`
```

Comment:
Isn't this then pretty much 'managed'?

If you do this then you cannot use a *lot* of functionality. TLS, scoping the route selection, etc.

I don't think creating Gateway from Service is a good idea, personally. The reverse is.

Reply:
I'll have to think on it some, there are definitely limitations here.